.github/instructions/uitests.instructions.md

```diff
@@ -199,6 +198,104 @@ public void SoftInputBehaviorTest()
 #endif
 ```
 
+## Running UI Tests Locally
+
+### Quick Test Execution (for rapid development)
+
+When developing and debugging a specific test:
+
+**Android:**
+1. Deploy the TestCases.HostApp:
+   ```bash
+   # Use local dotnet if available, otherwise use global dotnet
+   ./bin/dotnet/dotnet build src/Controls/tests/TestCases.HostApp/Controls.TestCases.HostApp.csproj -f net10.0-android -t:Run
+   # OR:
+   dotnet build src/Controls/tests/TestCases.HostApp/Controls.TestCases.HostApp.csproj -f net10.0-android -t:Run
+   ```
+
+2. Run your specific test:
+   ```bash
+   dotnet test src/Controls/tests/TestCases.Android.Tests/Controls.TestCases.Android.Tests.csproj --filter "FullyQualifiedName~Issue12345"
+   ```
+
+**iOS (3-step process):**
+
+**Step 1: Find iPhone Xs with highest API level**
+```bash
+# View all iPhone Xs devices with their iOS versions
+xcrun simctl list devices available | awk '/^--.*iOS/ {version=$0} /iPhone Xs/ {print version " -> " $0}'
+
+# Extract UDID of iPhone Xs with highest iOS version (last in list)
+UDID=$(xcrun simctl list devices available | grep "iPhone Xs" | tail -1 | sed -n 's/.*(\([A-F0-9-]*\)).*/\1/p')
+
+# Verify UDID was found
+if [ -z "$UDID" ]; then
+    echo "ERROR: No iPhone Xs simulator found. Please create an iPhone Xs simulator before running iOS tests."
+    exit 1
+fi
+
+echo "Using iPhone Xs with UDID: $UDID"
+```
+
+**Step 2: Build the iOS app**
+```bash
+# Use local dotnet if available, otherwise use global dotnet
+./bin/dotnet/dotnet build src/Controls/tests/TestCases.HostApp/Controls.TestCases.HostApp.csproj -f net10.0-ios
+# OR:
+dotnet build src/Controls/tests/TestCases.HostApp/Controls.TestCases.HostApp.csproj -f net10.0-ios
+```
+
+**Step 3: Boot simulator and install app (non-blocking)**
+```bash
+# Boot the simulator (will error if already booted, which is fine)
+xcrun simctl boot $UDID 2>/dev/null || true
+
+# Install the app to the simulator
+xcrun simctl install $UDID artifacts/bin/Controls.TestCases.HostApp/Debug/net10.0-ios/iossimulator-arm64/Controls.TestCases.HostApp.app
+
+# Verify simulator is booted
+xcrun simctl list devices | grep "$UDID"
```

Comment:
`xcrun simctl list devices` will list all devices, not only the booted one, so this will always succeed.